### PR TITLE
Remove bind method from Safemode::Scope

### PR DESCRIPTION
**Pull request by adamruzicka**
The bind method could be used to execute arbitrary system commands by leveraging that things passed into it were getting eval'd without any escaping. This should remove the issue completely by making bind unavailable.

### CODE REVIEW

File: lib/safemode.rb [hunk 1]
```diff
@@ -40,15 +40,16 @@ def find_jail_class(klass)
   
   class Box
     def initialize(delegate = nil, delegate_methods = [], filename = nil, line = nil)
-      @scope = Scope.new(delegate, delegate_methods)
```
Comment (ofedoren) [inline lib/safemode.rb:43]: `output` method below depends on `@scope` being available.

File: test/test_safemode_eval.rb [hunk 2]
```diff
@@ -80,6 +80,10 @@ def test_interpolated_xstr_should_raise_security
     assert_raise_security '"#{`ls -a`}"'
   end
 
+  def test_should_not_allow_access_to_bind
+    assert_raise_security "self.bind('an arg')"
```
Comment (ofedoren) [inline test/test_safemode_eval.rb:84]: Even though there is no longer `bind` method, is this possible to use `get_binding` directly instead?

**Review by ofedoren — COMMENTED** [review]
Thanks, @adamruzicka, looks sane, but I'm quite limited here, so just my two cents are:

File: lib/safemode.rb (hunk 1 above)
Comment (adamruzicka) [reply lib/safemode.rb:43]: Good catch

File: lib/safemode.rb [hunk 3]
```diff
@@ -39,20 +39,18 @@ def find_jail_class(klass)
...
       @filename = filename
       @line = line
     end    
 
     def eval(code, assigns = {}, locals = {}, &block)
       code = Parser.jail(code)
-      binding = @scope.bind(assigns, locals, &block)
-      result = Kernel.eval(code, binding, @filename || __FILE__, @line || __LINE__)
+      scope = Scope.new(@delegate, @delegate_methods, instance_vars: assigns, locals: locals, &block)
+      @output = Kernel.eval(code, scope.get_binding, @filename || __FILE__, @line || __LINE__)
```
Comment (ofedoren) [inline lib/safemode.rb:53]: Is this really the same as `scope.output`?

Comment (adamruzicka) [reply lib/safemode.rb]: Erm, no. /o\

**Review by ofedoren — APPROVED** [review]
Okay, now I have nothing to add :/

The weird thing is though that the tests didn't fail once...

**Comment by adamruzicka** [comment]
Alright, let's get this in

**Comment by Odilhao** [comment]
Hi @adamruzicka can we get this released into a new gem?

**Pull request merged**